ZOOKEEPER-2313 Refactor ZooKeeperServerBean and its subclasses

ravowlga123 opened the pull request:
As per ticket [ZOOKEEPER-2313](https://issues.apache.org/jira/browse/ZOOKEEPER-2313) following changes have been made:

- Added a constructor in ZookeeperServerBean and passed required parameter from its sub-classes (FollowerBean, LeaderBean, ObserverBean). 
- getName() methods from FollowerBean, LeaderBean, ObserverBean have been removed as well.
- ObserverBean class is moved to under quorum package.

Please do let me know if any additional changes are required.

maoling commented on zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/FollowerBean.java:
```diff
@@ -29,14 +29,10 @@
     public FollowerBean(Follower follower, ZooKeeperServer zks) {
-        super(zks);
+        super(zks, "FollowerServer_port" + zks.getClientPort());
         this.follower = follower;
```
`super(zks, "Follower")`;?

maoling commented on zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServerBean.java:
```diff
@@ -40,6 +40,12 @@ public ZooKeeperServerBean(ZooKeeperServer zks) {
         name = "StandaloneServer_port" + zks.getClientPort();
     }
 
```
-  `name = "StandaloneServer";` is OK, don't need the port? `StandaloneServer_port2181`? `StandaloneServer_port:2181`, `StandaloneServer_port_2181`:)
-  +1 for the moving of `ObserverBean`
- `ReadOnlyBean`  has been forgotten?

maoling commented:
It's could be better to use tool like `Jconsole` to observe the name of JMX bean to make sure this refactor is right.

ravowlga123 commented:
Thank You for the review @maoling. I have addressed your comments in the latest commit and also made the changes in ReadOnlyBean.

Regarding the test failures in previous build:

- org.apache.zookeeper.server.quorum.LeaderBeanTest.testGetName: Was because of different name given in previous commit. This should pass now.
- org.apache.zookeeper.test.ObserverMasterTest.testAdminCommands and org.apache.zookeeper.test.QuorumMajorityTest.testMajQuorums were not failing locally.

Please do let me know if you find something wrong with latest commit.
